### DISCUSSION TITLE
Can't parse even simple grammar

### DISCUSSION THREAD

**Discussion by konceptosociala** [pest-parser/pest#805] [Q&A] [answered]
**Describe the bug**
Parsing multiple line comments doesn't work

**To Reproduce**
1. Grammar: `multiline_comment = { "/*" ~ (ANY)* ~ "*/" }`
2. Any comment like 
```/* abc123 */```
Or
```
/*
*
* multiline
*
*/
```

**Expected behavior**
It must be parsed, but there is an error
```
Error: Error { variant: ParsingError { positives: [multiline_comment], negatives: [] }, location: Pos(0), line_col: Pos((1, 1)), path: None, line: "/* sample multiline comment */", continued_line: None }
```

**Comment by tomtau** [comment, marked as answer]
How about this rule: https://github.com/pest-parser/pest/blob/master/meta/src/grammar.pest#L164 ?

```
multiline_comment  = { "/*" ~ (multiline_comment | !"*/" ~ ANY)* ~ "*/" }
```

**Reply by konceptosociala** [reply]
Thanks, it works, but could you pls explain, what is wrong with my variant?

**Reply by tomtau** [reply]
PEG's matching is greedy and unambiguous (it will just deterministically pick one parse tree), so in your rule, it will just keep matching on `(ANY)*` until the end, and then fail to match `*/`.
